firebase-functions/src/main/java/com/google/firebase/functions/FirebaseFunctions.java

```diff
@@ -73,26 +74,40 @@
   private final String projectId;
 
   // The region to use for all function references.
-  private final String region;
+  @Nullable private final String region;
+
+  // A custom domain for the http trigger, such as "https://mydomain.com"
+  @Nullable private final String customDomain;
 
   // The format to use for constructing urls from region, projectId, and name.
   private String urlFormat = "https://%1$s-%2$s.cloudfunctions.net/%3$s";
 
+  // Allowed custom domain protocols.
+  private String[] customDomainSchemes = {"http", "https"};
+
   // Emulator settings
   @Nullable private EmulatedServiceSettings emulatorSettings;
 
   FirebaseFunctions(
       FirebaseApp app,
       Context context,
       String projectId,
-      String region,
+      String regionOrCustomDomain,
       ContextProvider contextProvider) {
     this.app = app;
     this.client = new OkHttpClient();
     this.serializer = new Serializer();
     this.contextProvider = Preconditions.checkNotNull(contextProvider);
     this.projectId = Preconditions.checkNotNull(projectId);
-    this.region = Preconditions.checkNotNull(region);
+
+    UrlValidator validator = new UrlValidator(customDomainSchemes);
```

Comment:
I think we can just use the standard library `URL` class and do `new URL(regionOrCustomDomain)` and then catch `MalformedUrlException`:
https://docs.oracle.com/javase/7/docs/api/java/net/URL.html#URL(java.lang.String)

Not quite as good as what you have but worth saving a whole dependency.

Reply:
Done